lib/devcycle-ruby-server-sdk/event_queue.rb

```diff
@@ -0,0 +1,128 @@
+require 'typhoeus'
+require 'sorbet-runtime'
+require 'concurrent-ruby'
+
+module DevCycle
+  class EventQueue
+    extend T::Sig
+
+    attr_accessor :api_client
+    attr_accessor :events_api_uri
+    attr_accessor :event_flush_interval_ms
+    attr_accessor :flush_event_queue_size
+    attr_accessor :flush_timer_task
+    attr_accessor :flush_in_progress
+    attr_accessor :local_bucketing
+    attr_accessor :logger
+    attr_accessor :max_event_queue_size
+    attr_accessor :sdk_key
+
+
+    sig { params(sdk_key: String, options: EventQueueOptions, local_bucketing: LocalBucketing).returns(NilClass) }
+    def initialize(sdk_key, options, local_bucketing)
+      @sdk_key = sdk_key
+      @events_api_uri = options.events_api_uri
+      @event_flush_interval_ms = options.event_flush_interval_ms || 10 * 1000
+      @api_client = api_client
+      @flush_in_progress = false
+      @logger = logger
+
+      if @event_flush_interval_ms < 500 || @event_flush_interval_ms > (60 * 1000)
+        raise ArgumentError.new('event_flush_interval_ms must be between 500ms and 1 minute')
+      end
+
+      @flush_event_queue_size = options.flush_event_queue_size || 1000
+      @max_event_queue_size = options.max_event_queue_size || 2000
+      chunk_size = options.event_request_chunk_size || 100
+      if @flush_event_queue_size > @max_event_queue_size
+        ArgumentError.new('flush_event_queue_size: #{flush_event_queue_size} must be' +
+          'larger than max_event_queue_size: #{max_event_queue_size}')
+      elsif @flush_event_queue_size < chunk_size || @max_event_queue_size < chunk_size
+        ArgumentError.new('event_request_chunk_size: #{chunk_size} must be smaller than' +
+          'flush_event_queue_size: #{flush_event_queue_size} and max_event_queue_size: ' +
+          '#{max_event_queue_size}')
+      elsif @flush_event_queue_size > 20000 || @max_event_queue_size > 20000
+        ArgumentError.new('flush_event_queue_size: #{flush_event_queue_size} or' +
+          'max_event_queue_size: #{max_event_queue_size} must be smaller than 20,000')
+      end
+
+      @flush_timer_task = Concurrent::TimerTask.new(
+        execution_interval: @event_flush_interval_ms / 1000,
+        run_now: true
+      ) {
+        flush_events
+      }
+      @flush_timer_task.execute
+
+      @local_bucketing = local_bucketing
+      @local_bucketing.init_event_queue(options)
+    end
+
+    sig { returns(NilClass) }
+    def flush_events
+      payloads = @local_bucketing.flush_event_queue
+      if payloads.size == 0
+        return
+      end
+      eventCount = payloads.reduce(0) { |sum, payload| sum + payload.eventCount }
+      payloads.each do |payload|
+        begin
+          response = Typhoeus.post(
+            @events_api_uri + 'v1/events/batch',
```

Comment:
We need to include the starting `/` as the URI shouldn't include the trailing `/`

Reply:
kk will fix